src/main/java/htsjdk/samtools/SAMRecord.java

```diff
@@ -1564,23 +1553,15 @@ public SAMTagAndValue(final String tag, final Object value) {
         return ret;
     }
 
-    Integer getIndexingBin() {
-        return mIndexingBin;
-    }
-
-    /**
-     * Used internally when writing BAMRecords.
-     * @param mIndexingBin c.f. http://samtools.sourceforge.net/SAM1.pdf
-     */
-    void setIndexingBin(final Integer mIndexingBin) {
-        this.mIndexingBin = mIndexingBin;
-    }
-
     /**
      * Does not change state of this.
      * @return indexing bin based on alignment start & end.
      */
     int computeIndexingBin() {
+        if (getAlignmentStart() > GenomicIndexUtil.BIN_GENOMIC_SPAN || getAlignmentEnd() > GenomicIndexUtil.BIN_GENOMIC_SPAN) {
+            throw new IllegalStateException("Read position too high for BAI bin indexing.");
+        }
+
         // regionToBin has zero-based, half-open API
         final int alignmentStart = getAlignmentStart()-1;
         int alignmentEnd = getAlignmentEnd();
```

Comment:
Move these two lines up so that getAlignmentStart() and getAlignmentEnd() aren't called twice.

Reply:
Done.